lighthouse-core/audits/treemap-data.js

```diff
@@ -0,0 +1,315 @@
+/**
+ * @license Copyright 2020 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview
+ * Creates treemap data for treemap app.
+ */
+
+const Audit = require('./audit.js');
+const JsBundles = require('../computed/js-bundles.js');
+const UnusedJavaScriptSummary = require('../computed/unused-javascript-summary.js');
+const ModuleDuplication = require('../computed/module-duplication.js');
+const NetworkRecords = require('../computed/network-records.js');
+const ResourceSummary = require('../computed/resource-summary.js');
+
+/**
+ * A collection of root nodes, grouped by type.
+ * @typedef {Record<string, RootNode[]>} TreemapData
+ */
+
+/**
+ * Ex: https://gist.github.com/connorjclark/0ef1099ae994c075e36d65fecb4d26a7
+ * @typedef RootNode
```

Comment:
this type feels a little more like `ScriptResource` or something  to me rather than `RootNode` and the `node` property of this type is the actual RootNode.

Am I misunderstanding?

Reply:
after looking at the resource summary bit I think I understand why this was done, but I'm a little unclear why we need that data like that, so maybe that's my root question :)

Reply:
I think trying to explain this right now is premature, and I'm actually having some trouble summarizing right now–most I can say is that this is naturally how the data evolved in the treemap viewer to distinguish named sets of tree data.

It would be nice to remove having both Root Node + Node, but let's wait until some of the treemap viewer PRs start coming in, and we have a bigger picture of everything.

ref https://github.com/GoogleChrome/lighthouse/compare/viewer-treemap-2#diff-ccf4c5b175a7960943aeb2a6460f9382

Reply:
alright sounds good, can we rename to something in the meantime just to indicate it's different from an actual node though? :) `RootNode(Entrypoint|Container|Resource|Label)` do any of those work?